contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -1,178 +1,299 @@
+            );
+            _setRecords(resolver, label, data);
         }
 
-        emit NameRegistered(name, label, owner, cost, expires);
-
-        // Refund any extra payment
-        if(msg.value > cost) {
-            payable(msg.sender).transfer(msg.value - cost);
+        if (reverseRecord) {
+            _setReverseRecord(name, resolver, msg.sender);
         }
+
+        emit NameRegistered(
+            name,
+            label,
+            owner,
+            msg.value - premiumCost,
+            premiumCost,
+            expires
+        );
     }
 
-    function renew(string calldata name, uint duration) external payable {
-        uint cost = rentPrice(name, duration);
-        require(msg.value >= cost);
+    function renew(string calldata name, uint256 duration)
+        external
+        payable
+        override
+    {
+        (uint256 basePrice, ) = rentPrice(name, duration);
+        require(msg.value >= basePrice);
 
         bytes32 label = keccak256(bytes(name));
-        uint expires = base.renew(uint256(label), duration);
+        uint256 expires = base.renew(uint256(label), duration);
 
-        if(msg.value > cost) {
-            payable(msg.sender).transfer(msg.value - cost);
+        if (msg.value > basePrice) {
+            payable(msg.sender).transfer(msg.value - basePrice);
         }
 
-        emit NameRenewed(name, label, cost, expires);
+        emit NameRenewed(name, label, basePrice, expires);
     }
 
     function setPriceOracle(PriceOracle _prices) public onlyOwner {
         prices = _prices;
         emit NewPriceOracle(address(prices));
     }
 
-    function setCommitmentAges(uint _minCommitmentAge, uint _maxCommitmentAge) public onlyOwner {
-        minCommitmentAge = _minCommitmentAge;
-        maxCommitmentAge = _maxCommitmentAge;
+    // function setCommitmentAges(
+    //     uint256 _minCommitmentAge,
+    //     uint256 _maxCommitmentAge
+    // ) public onlyOwner {
+    //     minCommitmentAge = _minCommitmentAge;
+    //     maxCommitmentAge = _maxCommitmentAge;
+    // }
+
+    function withdraw() public {
+        payable(owner()).transfer(address(this).balance);
     }
 
-    function withdraw() public onlyOwner {
-        payable(msg.sender).transfer(address(this).balance);        
+    function supportsInterface(bytes4 interfaceID)
+        external
+        pure
+        returns (bool)
+    {
+        return
+            interfaceID == INTERFACE_META_ID ||
+            interfaceID == COMMITMENT_CONTROLLER_ID;
     }
 
-    function supportsInterface(bytes4 interfaceID) external pure returns (bool) {
-        return interfaceID == INTERFACE_META_ID ||
-               interfaceID == COMMITMENT_CONTROLLER_ID ||
-               interfaceID == COMMITMENT_WITH_CONFIG_CONTROLLER_ID;
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes calldata
```

Comment:
`calledata` is not used anywhere

Reply:
I'm not sure if github is showing the right code snippet. But if it's L238, seems to be used